pkg/master/controller.go

```diff
@@ -117,6 +117,9 @@ func (c *Controller) UpdateKubernetesService() error {
 		if err := c.CreateMasterServiceIfNeeded("kubernetes", c.ServiceIP, servicePorts, serviceType); err != nil {
 			return err
 		}
+		if err := c.SetService("kubernetes", servicePorts, serviceType); err != nil {
```

Comment:
seems like we'd only want a single function `CreateOrUpdateMasterServiceIfNeeded`